### ISSUE TITLE
We need a TouchGestureRecognizer ASAP

### ISSUE THREAD

**Issue by davepruitt** [dotnet/maui#14297]
### Description

I am incredibly frustrated with .Net MAUI right now. Why is there no simple "touch gesture recognizer"?

I have a view on the screen, and all I want to do is see if it has been touched. _Not tapped_, because _tapped_ indicates there has been a _release of the finger_. I want to see if a _touch_ has happened. In other words a "press" as opposed to a "release".

It is frustratingly difficult to do this in MAUI right now, and the fact that this is not one of the first features implemented is inconceivable to me.

Right now, I only see a third party option which implements a "TouchEffect": https://github.com/FreakyAli/Maui.FreakyEffects

Not even the existing PointerGestureRecognizer contains a PointerPressed event, which would be helpful! (Similar issue here: https://github.com/dotnet/maui/issues/13346)

Also, my request would likely be nullified if MAUI actually had a true ContentButton - a button that supports _any_ content rather than just text or just an image. But that doesn't exist in MAUI either.

One of your main competitors, Flutter, contains both a **PointerDownEvent** and an **onTapDown** gesture (https://docs.flutter.dev/development/ui/advanced/gestures). I would have already switched this project to Flutter if the existing codebase hadn't already been written in Xamarin.

### Public API Changes

TouchGestureRecognizer. Simple as that.

### Intended Use-Case

Handling any kind of touch to any kind of view on the screen.

**Comment by mattleibow** [comment]
Duplicate of https://github.com/dotnet/maui/issues/13346

**Comment by jfversluis** [comment]
Hey @davepruitt sorry to hear about your frustration! I think there is multiple options here. The first one you already pointed out, there is a library already available. Why is that not something you could use?

The second is the Community Toolkit, although that doesn't really help you today, but things are going on there: https://github.com/CommunityToolkit/Maui/pull/1063 we hope to get that in sooner rather than later.

But reading between the lines I understand that you want this to be in the box. As much as I understand, I hope you also understand that a lot of people want to see their thing in the box and whatever that person is bringing up is most important to them. I'd love to get it all in, but there is a lot of ground to cover here.

That leaves us with two options: there are great options to reach into the platforms power and implement it in your own app code. Or, you can make friends and help yourself out by doing a pull request to .NET MAUI itself. That way, you're still spending time figuring this out, but as a result it's not just you who is benefiting but also everyone else. Win-win, the power of open-source.

It seems there is already a feature request open for this. In fact I think there might be multiple in different forms. Unfortunately these have not reached the top of our priority list yet.

I hope this helps and gives you some options to explore!

**Comment by davepruitt** [comment]
@jfversluis, to answer your initial question: Unfortunately that third-party option that I linked to doesn't really build properly. However, having said that, I ripped the code out of the third-party option, stuck it into my MAUI project, and edited it in such a way as to make it work/build properly.

But yes, I really firmly believe that things like this need to be "in the box". And it's not just this. MAUI feels like a really incomplete product right now, and I often feel like I am fighting MAUI to get it do to what I need, when my expectation is that MAUI should empower me or enable me to do what I need.

In all seriousness the only reason I'm not using Flutter on this project is because the codebase for what I'm working on right now was already done in Xamarin. Yall's competitor Flutter feels much more complete and, while it also has its own flaws, doesn't frustrate me every 5 minutes.

But frankly, handling a "press" should really not even be a discussion as to whether it should be "in the box" or a third-party option. It's one of the most basic user interactions.
